### ISSUE TITLE
[Bug]: Failed to insert data to Milvus: 'int' object has no attribute 'name'

### ISSUE THREAD

**Issue by tom729** [milvus-io/pymilvus#1324]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I reuse the reverse_image_search example to build a image similar search engine, I have succeed to insert about 50w image to the milvus, however, today I insert new data to milvus failed, I am not sure wether I did some configuraiton changes or some code changes. how can I find more useful info about the error, is there any other guy have the same problem?
the below is the insert function, which i rewrite to meet our needs.

### Steps/Code To Reproduce behavior

> ``def insert(self, collection_name, vectors,work_list,path_list,uid_list):
        # Batch insert vectors to milvus collection
        try:
            self.create_collection(collection_name)
            self.set_collection(collection_name)
            #LOGGER.debug(self.collection.schema)
            data = [vectors,work_list,path_list,uid_list]
            mr = self.collection.insert(data)
            LOGGER.debug(f"Insert vectors to Milvus in collection: {collection_name} with {len(vectors)} rows")
            ids = mr.primary_keys
            #self.collection.flush() 
            self.collection.load()
            LOGGER.debug("ids %s" %ids)
            return ids
        except Exception as e:
            LOGGER.error(f"Failed to insert data to Milvus: {e}")
            return(-1)``

_No response_

### Environment details

```markdown
- Hardware/Softward conditions (ubuntu, tencent cloud 4 CORE 16G memory):
- Method of installation (Docker):
- Milvus version (2.2.2)
- pymilvus version (2.2.2)
```

**Comment by tom729** [comment]
create colleciont codes: 
`    def create_collection(self, collection_name):
        # Create milvus collection if not exists
        try:
            if not self.has_collection(collection_name):
                field1 = FieldSchema(name="id", dtype=DataType.INT64, descrition="int64", is_primary=True, auto_id=True)
                field2 = FieldSchema(name="embedding", dtype=DataType.FLOAT_VECTOR, descrition="float vector",
                                     dim=VECTOR_DIMENSION, is_primary=False)
                field3 = FieldSchema(name="work_id", dtype=DataType.INT64, descrition="work id", is_primary=False,auto_id=False)
                field4 = FieldSchema(name='path', dtype=DataType.VARCHAR, description='path to image', max_length=200, is_primary=False, auto_id=False)
                field5 = FieldSchema(name="uid", dtype=DataType.INT64, descrition="user id", is_primary=False,auto_id=False)
                schema = CollectionSchema(fields=[field1, field2,field3,field4,field5], description="mepai reverse image search collection")
                self.collection = Collection(name=collection_name,data=None, schema=schema)
                LOGGER.debug(f"Create Milvus collection: {collection_name}")
                self.create_index(collection_name) #added by tom
            else:
                self.set_collection(collection_name)
            return "OK"
        except Exception as e:
            LOGGER.error(f"Failed to create_collection to Milvus: {e}")
            sys.exit(1)`

**Comment by tom729** [comment]
the data to be inserted
`data [[[0.003984056, 0.05035976, 0.0035724803, 0.024251372, 0.012805485, 0.0, 0.20567413, 0.06280334, 0.0, 0.007527845, 0.0, 0.009578028, 0.06537565, 0.0015505916, 0.0, 0.029210553, 0.08572697, 0.07296065, 0.10423706, 0.06479557, 0.00016332873, 0.0005436356, 0.021337505, 0.11780061, 1.8573557e-05, 0.007788484, 0.0, 0.0, 0.0034552256, 0.0003037396, 0.0, 0.017227395, 0.12302162, 0.080519386, 0.008588848, 0.001594326, 0.0, 0.0022909122, 0.07437763, 0.002085234, 0.0049128644, 0.009436793, 0.04508261, 0.03350333, 0.0, 0.0, 0.014462492, 0.0, 0.009703101, 0.049374476, 0.0035903952, 0.058247454, 0.0011420114, 0.0138136465, 0.018960508, 0.017010711, 0.002970688, 0.060021922, 0.087262236, 0.0, 0.0012555167, 0.046908338, 0.014382234, 0.0507905, 0.0047653737, 0.0, 0.0, 0.0064109424, 0.00032438096, 0.0, 0.00080815377, 0.026217148, 0.0, 0.010353518, 0.004991016, 0.08405566, 0.16280639, 0.07468212, 0.043359764, 0.0, 0.007934334, 0.04986558, 0.11259405, 0.06668631, 0.04409146, 0.0, 0.01633408, 0.0, 0.13158669, 0.0, 0.0, 0.06308784, 0.057487786, 0.00040440162, 0.14139564, 0.0, 0.0, 0.031943258, 0.01438274, 0.008133374, 0.00066351576, 0.0, 0.0, 0.015506038, 0.004599146, 0.015863629, 0.23873715, 0.0020086854, 0.03241497, 0.0, 0.02100632, 0.011128495, 0.0012910061, 0.0150160035, 0.011986496, 0.00017275073, 0.026956057, 0.031788018, 0.031039633, 0.044635307, 0.00011598573, 0.024265882, 0.010104212, 0.0, 0.014292167, 0.007132172, 0.0, 0.0125404745, 0.043985803, 0.023217553, 0.0056506963, 0.006951858, 0.049252994, 0.1896605, 0.009707178, 0.0654714, 0.0019808046, 0.027309336, 0.004400177, 0.017826403, 0.042552434, 0.058321528, 0.014891974, 0.011432424, 0.06476304, 0.006800652, 0.030041484, 0.0, 0.00037660124, 0.012896231, 0.045910228, 0.019383285, 0.0, 0.040070675, 0.050089687, 0.008607105, 0.00464721, 0.0, 0.0, 0.01079142, 0.0005718102, 0.024374522, 0.0, 0.016067471, 0.009117074, 0.00013355278, 0.0009458347, 0.0035210538, 0.0, 0.01916649, 0.020891963, 0.024629703, 0.0013206033, 0.03577105, 0.009717608, 0.0025785516, 0.074726276, 0.07524932, 0.14354764, 0.0026401097, 0.014049035, 0.0667825, 0.08519748, 0.0, 0.035256103, 0.019407876, 0.0, 0.026124647, 0.070833676, 0.0025549757, 0.020842817, 0.0064744097, 0.01777372, 0.003000723, 0.02818301, 0.03244788, 0.14925152, 0.016457627, 0.00065684924, 0.021579105, 0.07460491, 0.0, 0.026042337, 0.08290257, 0.0, 0.0015368989, 0.0532502, 0.0033288307, 0.002450509, 0.0, 0.2071008, 0.0011872732, 0.0021029876, 0.011937993, 0.018547501, 0.017532837, 0.0, 0.0, 0.04161505, 0.0, 0.0, 0.024442662, 0.017388823, 0.0, 0.07437103, 0.115296155, 0.018569434, 0.053693347, 0.02980576, 0.0015618111, 0.0, 0.013929048, 0.031696912, 0.034315605, 0.0015530312, 0.008796366, 0.08699985, 0.03232618, 0.027799886, 0.005852861, 0.0, 0.058900114, 0.008075089, 0.03841524, 0.025615517, 0.030076342, 0.024190005, 0.066724345, 0.003536829, 0.069756545, 0.009384953, 0.004649262, 0.0, 0.010075036, 0.0, 0.017632304, 0.04799124, 0.001377099, 0.010750085, 0.0, 0.0066253855, 0.0075774626, 0.0119697545, 0.011490415, 0.004015858, 0.06597905, 0.010978176, 0.0040536476, 0.0063205967, 0.028223446, 0.027615504, 0.007764961, 0.0954049, 0.045936488, 0.0, 0.0, 0.0134522505, 0.003716552, 0.058931522, 0.004556165, 0.1479262, 0.0075736316, 0.030045567, 0.034583844, 0.010594138, 0.0012902603, 0.0047344854, 0.0025530078, 0.025112525, 0.018681135, 0.0, 0.06395489, 0.060102295, 0.0039721993, 0.0070763985, 3.5505636e-05, 0.0, 0.011829462, 0.00355062, 0.0, 0.01292252, 0.0012218993, 0.053101417, 0.0, 0.0035283554, 0.017344663, 0.0033587003, 0.011984821, 0.01602541, 0.04746076, 0.07728138, 0.0, 0.013932722, 0.0, 0.0012883694, 0.0066818073, 0.026422184, 0.00067642686, 0.050224595, 0.024135867, 0.04664738, 0.052487615, 0.0, 0.08686855, 0.009662133, 0.0, 0.0, 0.040999424, 0.01960686, 0.07318281, 0.014305945, 0.004119341, 0.0, 0.098494835, 0.03439365, 0.07280488, 0.0074880444, 0.010687445, 0.001466668, 0.02913183, 0.0005102688, 0.034190208, 0.00021371938, 0.053076997, 0.01984739, 0.010501185, 0.012184148, 0.0017006943, 0.0076079895, 0.01761315, 0.0014281602, 0.0, 0.011378986, 0.0, 0.011985844, 0.029300047, 0.002037424, 0.056578822, 0.012654261, 0.0047304705, 0.0, 0.0, 0.0, 0.012036085, 0.0034270876, 0.14365877, 0.024884801, 0.0, 0.0, 0.06818007, 0.009951498, 0.16996957, 0.03561168, 0.0011618551, 0.006068176, 0.02141328, 0.023133205, 0.0018640577, 0.009834625, 0.0, 0.028228125, 0.02721398, 0.020386267, 0.0, 0.044452384, 0.002305128, 0.0019697396, 0.00016998104, 0.009476683, 0.0, 0.023151027, 0.003295364, 0.0, 0.08732291, 0.007284708, 0.022113444, 0.054372687, 0.040609963, 0.0, 0.0, 0.038845494, 0.14208393, 0.0, 0.04437751, 0.0024238, 0.015084556, 0.05031972, 0.055528186, 0.018027144, 0.0014771337, 0.07221192, 0.018938383, 0.023220895, 0.0004113873, 0.028763842, 0.0020553337, 0.012294775, 0.0, 0.0, 0.026467634, 0.0018558751, 0.005871507, 0.019867236, 0.010094523, 0.0, 0.041886855, 0.056452762, 0.0038403345, 0.005691896, 0.009104954, 0.027221788, 0.009362086, 0.0, 0.07092444, 0.017654235, 0.040950246, 0.0006828864, 0.0042928685, 0.034071032, 0.019660171, 0.077978, 0.016750112, 0.008584821, 0.0, 0.12543567, 0.115137644, 0.003299817, 0.033715915, 0.08045286, 0.034845475, 0.18582067, 0.003173295, 0.05800497, 0.0, 0.0025099632, 0.043932933, 0.049812615, 0.0069951606, 0.019554032, 0.017394237, 0.0018557728, 0.0, 0.013428026, 0.009739373, 0.032863222, 0.015199916, 0.038648114, 0.0, 0.0125207715, 0.02931585, 0.0018382544, 0.12441031, 0.013526848, 0.0, 0.0105795795, 4.899803e-05, 0.00088777975, 0.0022360096, 0.0, 0.03927257, 0.025649754, 0.023292476, 0.032710165, 0.008697723, 0.0029202183, 0.0001508094, 0.0, 0.011784591, 0.03342348, 0.012244206, 0.0, 0.09407471, 0.00062895054, 0.10375275, 0.0, 0.072023585, 0.0, 0.02429903, 0.0, 0.0027633125, 0.0074742516, 0.00065687625, 0.010473168, 0.035232287, 0.054251943, 0.023808474, 0.0, 0.046690673, 0.0, 0.15122147, 0.010169485, 0.0]], ['15755403'], ['https://xxx.com/app/works/105149/2023-01-11/w_63be653c4643b/963be653c8aa8c.jpg'], ['105149']]`

**Comment by XuanYang-cn** [comment]
Format the code

```python
# create colleciont codes: 
def create_collection(self, collection_name): # Create milvus collection if not exists 
    try: 
        if not self.has_collection(collection_name): 
            field1 = FieldSchema(name="id", dtype=DataType.INT64, descrition="int64", is_primary=True, auto_id=True) 
            field2 = FieldSchema(name="embedding", dtype=DataType.FLOAT_VECTOR, descrition="float vector", dim=VECTOR_DIMENSION, is_primary=False) 
            field3 = FieldSchema(name="work_id", dtype=DataType.INT64, descrition="work id", is_primary=False,auto_id=False) 
            field4 = FieldSchema(name='path', dtype=DataType.VARCHAR, description='path to image', max_length=200, is_primary=False, auto_id=False) 
            field5 = FieldSchema(name="uid", dtype=DataType.INT64, descrition="user id", is_primary=False,auto_id=False) 
            schema = CollectionSchema(fields=[field1, field2,field3,field4,field5], description="mepai reverse image search collection") 
            self.collection = Collection(name=collection_name,data=None, schema=schema) 
            LOGGER.debug(f"Create Milvus collection: {collection_name}") 
            self.create_index(collection_name) #added by tom 
        else: 
            self.set_collection(collection_name) 
            return "OK" 
    except Exception as e: 
        LOGGER.error(f"Failed to create_collection to Milvus: {e}") 
        sys.exit(1)`

```

```python
def insert(self, collection_name, vectors,work_list,path_list,uid_list):         # Batch insert vectors to milvus collection
     try:
         self.create_collection(collection_name)
         self.set_collection(collection_name)
         #LOGGER.debug(self.collection.schema)
         data = [vectors,work_list,path_list,uid_list]
         mr = self.collection.insert(data)
         LOGGER.debug(f"Insert vectors to Milvus in collection: {collection_name} with {len(vectors)} rows")
         ids = mr.primary_keys 
         #self.collection.flush() 
         self.collection.load()
         LOGGER.debug("ids %s" %ids) 
         return ids         
    except Exception as e:
          LOGGER.error(f"Failed to insert data to Milvus: {e}")
          return(-1)
```

**Comment by tom729** [comment]
@XuanYang-cn is there any wrong of the above codes?

**Comment by marlonfan** [comment]
> @XuanYang-cn is there any wrong of the above codes?

```
Traceback (most recent call last):
  File "D:\code\py\py-openai-embedding-search\parse.py", line 26, in <module>
    conn.insert(data)
  File "D:\code\py\py-openai-embedding-search\.venv\lib\site-packages\pymilvus\orm\collection.py", line 426, in insert
    check_insert_data_schema(self._schema, data)
  File "D:\code\py\py-openai-embedding-search\.venv\lib\site-packages\pymilvus\orm\schema.py", line 327, in check_insert_data_schema
    raise DataNotMatchException(message=f"The data type of field {y.name} doesn't match, expected: {y.dtype.name}, got {x.dtype.name}")
AttributeError: 'int' object has no attribute 'name'
```

**Comment by marlonfan** [comment]
i reproduced the question

**Comment by marlonfan** [comment]
Hi, is anyone there? ✋

**Comment by XuanYang-cn** [comment]
@tom729 You inserted string data to int64 field "work_id"

**Comment by XuanYang-cn** [comment]
#1333 should fix the error message, will be available at pymilvus 2.3.0 & 2.2.4

This error is raised when the data type provided by insert mismatches with schema, Please double check the type of insert data.

This error message is buggy and unable to display informative error information, this won't be a problem after #1333
